src/Data/Repositories/WalletRepository.cs

```diff
@@ -385,5 +386,128 @@ public async Task<(bool, string?)> FinaliseWallet(Wallet selectedWalletToFinalis
+                var (strategyBase, tuples) = WalletParser.ParseOutputDescriptor(outputDescriptor, CurrentNetworkHelper.GetCurrentNetwork());
+
+                var keys = tuples.Select(x => new Key
+                {
+                    Id = 0,
+                    CreationDatetime = DateTimeOffset.UtcNow,
+                    Name = "Imported key from output descriptor",
+                    XPUB = x.Item1.ToString() ?? throw new InvalidOperationException(),
+                    Description =null,
+                    MasterFingerprint = x.Item2.MasterFingerprint.ToString(),
+                    Path = x.Item2.KeyPath.ToString(),
+                    IsBIP39ImportedKey = false,
+                    UserId = userId,
+                    
+                }).ToList();
+                
+                Wallet? wallet;
+                //if singlesig
+                if (strategyBase is DirectDerivationStrategy)
+                {
+                    wallet = new Wallet
+                    {
+                        Id = 0,
+                        CreationDatetime = DateTimeOffset.UtcNow,
+                        Name = name,
+                        Description = description,
+                        IsArchived = false,
+                        IsCompromised = false,
+                        IsFinalised = true,
+                        WalletAddressType = WalletAddressType.NativeSegwit,
+                        IsHotWallet = false,
+                        IsBIP39Imported = false,
+                        ImportedOutputDescriptor = outputDescriptor,
+                        BIP39Seedphrase = null,
+                        InternalWalletSubDerivationPath = null,
+                        InternalWalletMasterFingerprint = null,
+                        Keys = keys,
+                        InternalWalletId = null,
+                        MofN = 1,
+
+
+                    };
+                }
+                else if (strategyBase is P2WSHDerivationStrategy p2WshDerivationStrategy && p2WshDerivationStrategy.Inner is MultisigDerivationStrategy multisigDerivationStrategy)
+                {
+                    
+                    wallet = new Wallet
+                    {
+                        Id = 0,
+                        CreationDatetime = DateTimeOffset.UtcNow,
+                        Name = name,
+                        Description = description,
+                        IsArchived = false,
+                        IsCompromised = false,
+                        IsFinalised = true,
+                        WalletAddressType = WalletAddressType.NativeSegwit,
+                        IsHotWallet = false,
+                        IsBIP39Imported = false,
+                        ImportedOutputDescriptor = outputDescriptor,
+                        BIP39Seedphrase = null,
+                        Keys = keys,
+                        MofN = multisigDerivationStrategy.RequiredSignatures
+                    };
+                }
```

Comment:
Having a factory might be good for these cases so there its less likely to manual errors for some combination of values! No need to refactor now, just a comment :)